File: 0027-remove-element/0027-remove-element.go
func removeElement(nums []int, val int) int {
    res := 0
    l, r := 0, 0
    for r < len(nums) {
        if nums[r] != val {
            nums[l] = nums[r]
            l += 1
            res += 1
        }

        r += 1
    }

    return res
}

File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        indexCount = 0
        for i, value in enumerate(nums):
            if val != value:
                nums[indexCount], nums[i] = nums[i], nums[indexCount]
                indexCount+=1

        return indexCount 
        